redisinsight/api/src/modules/database-recommendation/scanner/recommendation.provider.ts

```diff
@@ -0,0 +1,31 @@
+import { Injectable } from '@nestjs/common';
+import { IRecommendationStrategy } from
+  'src/modules/database-recommendation/scanner/recommendation.strategy.interface';
+import { RECOMMENDATION_NAMES } from 'src/constants';
+import { DatabaseService } from 'src/modules/database/database.service';
+import {
+  DefaultRecommendationStrategy,
+  RediSearchStrategy,
+} from 'src/modules/database-recommendation/scanner/strategies';
+
+@Injectable()
+export class RecommendationProvider {
+  private strategies: Map<string, IRecommendationStrategy> = new Map();
+
+  constructor(
+    databaseService: DatabaseService,
+  ) {
+    this.strategies.set('default', new DefaultRecommendationStrategy());
+    this.strategies.set(RECOMMENDATION_NAMES.SEARCH_STRING, new RediSearchStrategy(databaseService));
+  }
```

Comment:
What is RecommendationProvider and RecommendationStrategy?
if each recommendation type will have own strategy to determine if we must create recommendation - I'm totally fine but I don't understand what is default strategy here? 
We should throw an error (or silently fail) if there is no such recommendation type, no?

Reply:
Also why **default** is outside of RECOMMENDATION_NAMES?